frontend/resourceadm/components/AccessListMembers/AccessListMembers.tsx

```diff
@@ -0,0 +1,240 @@
+import { FieldWrapper } from '../FieldWrapper';
+import { useRemoveAccessListMemberMutation } from 'resourceadm/hooks/mutations/useRemoveAccessListMemberMutation';
+import { useAddAccessListMemberMutation } from 'resourceadm/hooks/mutations/useAddAccessListMemberMutation';
+import { AccessListMemberRow } from './AccessListMemberRow';
+import { useDebounce } from 'react-use';
+import { usePartiesRegistryQuery } from 'resourceadm/hooks/queries/usePartiesRegistryQuery';
+import { useSubPartiesRegistryQuery } from 'resourceadm/hooks/queries/useSubPartiesRegistryQuery';
+import { getPartiesQueryUrl } from 'resourceadm/utils/urlUtils';
+import { StudioSpinner } from '@studio/components';
+import { PlusIcon, PlusCircleIcon, MinusCircleIcon } from '@studio/icons';
+
+const COLUMN_SPAN = 100;
+const PARTY_SEARCH_TYPE = 'PARTY';
+const SUBPARTY_SEARCH_TYPE = 'SUBPARTY';
+
+export interface AccessListMembersProps {
+  org: string;
+  env: string;
+  list: AccessList;
+}
+
+export const AccessListMembers = ({ org, env, list }: AccessListMembersProps): React.ReactNode => {
+  const { t } = useTranslation();
+
+  const [listItems, setListItems] = useState<AccessListMember[]>(list.members ?? []);
+  const [isAddMode, setIsAddMode] = useState<boolean>((list.members ?? []).length === 0);
+  const [isSubPartySearch, setIsSubPartySearch] = useState<boolean>(false);
+  const [searchText, setSearchText] = useState<string>('');
+  const [searchUrl, setSearchUrl] = useState<string>('');
+  useDebounce(() => setSearchUrl(getPartiesQueryUrl(searchText, isSubPartySearch)), 500, [
+    searchText,
+    isSubPartySearch,
+  ]);
+
+  const { mutate: removeListMember } = useRemoveAccessListMemberMutation(org, list.identifier, env);
+  const { mutate: addListMember } = useAddAccessListMemberMutation(org, list.identifier, env);
+
+  const { data: partiesSearchData, isLoading: isLoadingParties } = usePartiesRegistryQuery(
+    !isSubPartySearch ? searchUrl : '',
+  );
+  const { data: subPartiesSearchData, isLoading: isLoadingSubParties } = useSubPartiesRegistryQuery(
+    isSubPartySearch ? searchUrl : '',
+  );
+
+  // add member
+  const handleAddMember = (memberToAdd: AccessListMember): void => {
+    addListMember(memberToAdd.orgNr);
+    setListItems((old) => [...old, memberToAdd]);
+  };
+
+  // remove member
+  const handleRemoveMember = (memberIdToRemove: string): void => {
+    removeListMember(memberIdToRemove);
+    setListItems((old) => old.filter((x) => x.orgNr !== memberIdToRemove));
+  };
+
+  const resultData = partiesSearchData ?? subPartiesSearchData ?? undefined;
+
+  const renderPageButton = (href: string, label: string, isDisabled: boolean): React.ReactNode => {
+    return (
+      <Button
+        size='small'
+        variant='tertiary'
+        disabled={isDisabled}
+        onClick={() => setSearchUrl(href)}
+      >
+        {t(label)}
+      </Button>
+    );
+  };
+
+  return (
```

Comment:
I think this component has a lot going on. Is there any possibility to separate into smaller components? 😄

Reply:
Will move the pagingation to a separate component